Sources/Base/Statements.swift

```diff
@@ -75,6 +75,15 @@ class VarAssignDecl: Decl {
     superAttrs["mutable"] = mutable
     return superAttrs
   }
+
+  override var type: DataType {
+    set {
+      super.type = newValue
+    }
+    get {
+      return self.rhs?.type ?? super.type
```

Comment:
"Done" in the sense of "an assign expression's type was being 'deduced' by Sema to enable this upcast, but should actually be a function of the right hand side with a coercion inserted by the typechecker".